.github/workflows/ci-check-bundle-size.yml

```
@@ -0,0 +1,88 @@
+name: Check Binaries binary size
+
+on:
+  push:
+    branches: [main]
+
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+# See https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions
+permissions:
+  contents: read
+
+jobs:
+  build-and-check:
```

let's move this as yet another job into .github/workflows/ci-lint-checks.yaml so that it runs as part of the overall lint group.
```suggestion
  binary-size-check:
```